coalib/results/Diff.py

```diff
@@ -20,18 +20,26 @@ def __init__(self, file_list=None,
         Creates an empty diff for the given file.
 
         :param file_list: The original (unmodified) file as a list of its
-                          lines.
+                          lines. It has to be provided if ``create`` is not set,
+                          else an error will be raised.
         :param rename:    False or str containing new name of file.
         :param delete:    True if file is set to be deleted.
         :param create:    False or str containing the name of the file to be
                           created.
+        :raises ValueError: If ``file_list`` is not provided and ``create`` is
+                            not set.
         """
         self._changes = {}
-        self._file = [] if create else file_list
+        self._file = file_list
```

Comment:
I assign file_list, so that I can check if it is none, if it is, then only raise the ValueError, and if it is not raising it, everything is good and then assign it as `[]` if create